### PR TITLE
Setup github actions for testing and remove TravisCI

### PR DESCRIPTION
**Pull request by zacaudette**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (tkuhlman) [inline .github/workflows/test.yml:7]: Lets run it on all pull requests

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-$GITHUB_REPOSITORY-${{ hashFiles('**/go.mod') }}
+      - name: install dependencies
+        run: go mod download
+      - name: go vet
+        run: go vet ./...
+      - name: go test race
+        run: go test -v -race -parallel 4 -coverprofile=covprofile -covermode=atomic ./...
+      - name: install goveralls
```
Comment (tkuhlman) [inline .github/workflows/test.yml:29]: I'm fine with ditching coveralls but that can be discussed and made another PR

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.5'
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-$GITHUB_REPOSITORY-${{ hashFiles('**/go.mod') }}
+      - name: install dependencies
+        run: go mod download
```
Comment (tkuhlman) [inline .github/workflows/test.yml:24]: I don't think this is needed at all, go test will pull modules it needs.

**Review by tkuhlman — COMMENTED** [review]
👍🏼 to ditching Travis. Some nice follow ups to this would be automatic tagging.

File: .github/workflows/test.yml (hunk 2 above)
Comment (zacaudette) [reply .github/workflows/test.yml:29]: Agreed, with the mgspack, avro, and some of the generated code there is a lot that won't be tested.

**Review by tkuhlman — APPROVED** [review]
